spark/src/main/java/com/netflix/iceberg/spark/source/Writer.java

```diff
@@ -371,7 +385,8 @@ private void closeCurrent() throws IOException {
         this.currentAppender = null;
 
         DataFile dataFile = DataFiles.builder(spec)
-            .withInputFile(currentFile.toInputFile())
+            .withInputFile(currentFile.encryptingOutputFile().toInputFile())
+            .withEncryptionKeyMetadata(currentFile.keyMetadata())
```

Comment:
Similar alternative to the unpartitioned case, consider adding `withEncryptedOutputFile` that accesses the underlying `toInputFile` and `keyMetadata`.